### PR TITLE
added `irrchars` and `classes` methods for tables

### PR DESCRIPTION
**Pull request by SoongNoonien**
@fingolfin Is this what you meant in #16?

### CODE REVIEW

**Comment by SoongNoonien** [comment]
The names are not ideal, see #41. But I don't know better names right now.

File: src/Show.jl [hunk 1]
```diff
@@ -49,6 +49,26 @@ function irrchartypes(t::Table)
...
+```jldoctest
+julia> g=genchartab(\"GL2\");
+
+julia> irrchars(g)
+q^2 - 1
+
+```
+"""
+function irrchars(t::Table)
+	return sum(nrchars.(Ref(t), 1:irrchartypes(t)))
```
Comment (fingolfin) [inline src/Show.jl:69]: TIL one can use `Ref` like that in a broadcast, nice :-)

File: src/Show.jl [hunk 2]
```diff
@@ -49,6 +49,26 @@ function irrchartypes(t::Table)
 	return t.table.original_m
 end
 
+"""
+    irrchars(t::Table)
```
Comment (fingolfin) [inline src/Show.jl:53]: I would just call this `nrchars` as well.

And `irrchartypes` should be nirrchartypes` (or rather in OSCAR style: `n_irrchartypes` -- or even `number_of_irreducible_character_types` or whatever... )

File: src/Show.jl [hunk 3]
```diff
@@ -67,6 +87,24 @@ function classtypes(t::Table)
 	size(t.table, 2)
 end
 
+"""
+    classes(t::Table)
```
Comment (fingolfin) [inline src/Show.jl:91]: As above I'd just call this `nrclasses` as well

And eventually rename both to `n_classes`, and `classtypes` to `n_classtypes` -- but please not in this PR, this needs to be carefully coordinated.

**Comment by fingolfin** [comment]
I've resolved the conflicts and updated this to look like I meant.

But I am now not so sure this is a good idea. Isn't it true that some of the character/class types may not be disjoint? At least that's how I read the CHEVIE paper from 1996. ...

Anyway I don't need this right now, I can try to talk with e.g. Gunter Malle about this (I want to do that anyway for a bunch of things in this code)

**Comment by SoongNoonien** [comment]
> But I am now not so sure this is a good idea. Isn't it true that some of the character/class types may not be disjoint? At least that's how I read the CHEVIE paper from 1996. ...

I don't know, I thought that this is not the case if we consider the parameter exceptions which is done here.

File: src/Show.jl [hunk 4]
```diff
@@ -68,6 +68,26 @@ function irrchartypes(t::Table)
...
+# Examples
+```jldoctest
+julia> g=genchartab(\"GL2\");
+
+julia> nrirrchars(g)
+q^2 - 1
+
+```
+"""
+function nrirrchars(t::Table)
```
Comment (SoongNoonien) [inline src/Show.jl:87]: Is this function exported?

Comment (fingolfin) [reply src/Show.jl:88]: Oops! Now it is.

**Comment by SoongNoonien** [comment]
Should we wait with the merge until it is clear if we are counting some characters twice?

**Comment by SoongNoonien** [comment]
I've talked to Gunter Malle today and he thinks this should be fine, so now characters/classes are be counted multiple times.

**Pull request merged**